packages/teams-js/src/private/hostEntity.ts

```diff
@@ -0,0 +1,301 @@
+import { sendMessageToParentAsync } from '../internal/communication';
+import { ensureInitialized } from '../internal/internalAPIs';
+import { ApiName, ApiVersionNumber, getApiVersionTag } from '../internal/telemetry';
+import { ErrorCode, SdkError } from '../public';
+import { TabInformation, TabInstance } from '../public/interfaces';
+import { runtime } from '../public/runtime';
+
+/**
+ * v2 APIs telemetry file: All of APIs in this capability file should send out API version v2 ONLY
+ */
+const hostEntityTelemetryVersionNumber: ApiVersionNumber = ApiVersionNumber.V_2;
+
+/**
+ * @hidden
+ * @internal
+ * @beta
+ * Limited to Microsoft-internal use
+ *
+ * This capability allows an app to associate apps with a host entity, such as a Teams channel or chat, and configure them as needed.
+ */
+export namespace hostEntity {
+  export enum AppTypes {
+    edu = 'EDU',
+  }
+
+  export interface HostEntityIds {
+    /**
+     * Id of the host entity like channel, chat and meetings
+     */
+    threadId: string;
+
+    /**
+     * Id of message in which channel meeting is created
+     */
+    parentMessageId?: string;
+  }
+
+  /**
+   * @hidden
+   * @internal
+   * @beta
+   * Limited to Microsoft-internal use
+   *
+   * CRUD operations for tabs associated with apps
+   */
+  export namespace tab {
+    /**
+     * @hidden
+     * @internal
+     * @beta
+     * Limited to Microsoft-internal use
+     *
+     * Launches host-owned UI that lets a user select an app, installs it if required,
+     * runs through app configuration if required, and then associates the app with the threadId provided
+     *
+     * @param hostEntityIds Ids of the host entity like channel, chat or meeting
+     *
+     * @param appTypes What type of applications to show the user. If EDU is passed as appType, only apps supported by EDU tenant are shown.
+     * If no value is passed, all apps are shown.
+     *
+     * @returns The TabInstance of the newly associated app
+     *
+     * @throws Error if host does not support this capability, library as not been initialized successfully, input parameters are invalid, user cancels operation or installing
+     * or configuring or adding tab fails
+     */
+    export function addAndConfigure(hostEntityIds: HostEntityIds, appTypes?: AppTypes[]): Promise<TabInstance> {
```

Comment:
>appTypes?: AppTypes[] [](http://example.com/codeflow?start=66&length=21)

What's the semantic difference between not specifying this parameter and providing an empty array?

Reply:
Its the same. If no appType is specified in the array, we show all the apps. I'll add that to the comments.